run/idp-sql/test/system.test.js

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+const admin = require('firebase-admin');
+const assert = require('assert');
+const path = require('path');
+const supertest = require('supertest');
+const got = require('got');
+const uuid = require('short-uuid');
+const {execSync} = require('child_process');
+
+describe('System Tests', () => {
+
+  const {GOOGLE_CLOUD_PROJECT} = process.env;
+  if (!GOOGLE_CLOUD_PROJECT) {
+    throw Error('"GOOGLE_CLOUD_PROJECT" env var not found.');
+  }
+
+  const {CLOUD_SQL_CONNECTION_NAME} = process.env;
+  if (!CLOUD_SQL_CONNECTION_NAME) {
+    throw Error('"CLOUD_SQL_CONNECTION_NAME" env var not found.');
+  }
+
+  const {DB_PASSWORD} = process.env;
+  if (!DB_PASSWORD) {
+    throw Error('"DB_PASSWORD" env var not found.');
+  }
+
+  const {DB_NAME, DB_USER, SAMPLE_VERSION} = process.env;
+
+  let {SERVICE_NAME} = process.env;
+  if (!SERVICE_NAME) {
+    console.log('"SERVICE_NAME" env var not found. Defaulting to "idp-sql"');
+    SERVICE_NAME = "idp-sql";
+  }
+
+  let BASE_URL, ID_TOKEN;
+
+  before(() => {
+    const buildCmd = `gcloud builds submit --project ${GOOGLE_CLOUD_PROJECT} ` +
+      `--config ./test/e2e_test_setup.yaml ` +
+      `--substitutions _SERVICE=${SERVICE_NAME},_DB_PASSWORD=${DB_PASSWORD}` +
+      `,_CLOUD_SQL_CONNECTION_NAME=${CLOUD_SQL_CONNECTION_NAME}`;
+    if(SAMPLE_VERSION) buildCmd + `,_VERSION=${SAMPLE_VERSION}`;
+    if(DB_USER) buildCmd + `,_DB_USER=${DB_USER}`;
+    if(DB_NAME) buildCmd + `,_DB_NAME=${DB_NAME}`;
+
+    console.log('Starting Cloud Build...');
+    execSync(buildCmd);
+    console.log('Cloud Build completed.');
+
+    const url = execSync(
+      `gcloud run services describe ${SERVICE_NAME} --project=${GOOGLE_CLOUD_PROJECT} ` +
+      `--platform=managed --region=us-central1 --format='value(status.url)'`);
```

Comment:
nit: Since this file "owns" how the testing will work and the Cloud Build file is configurable, ideally the build substitutions will be set from this file instead of rely on synchronized defaults.